controllers/kustomization_controller.go

```diff
@@ -175,7 +175,10 @@ func (r *KustomizationReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Finalise the reconciliation and report the results.
 	defer func() {
-		retErr = r.finalizeStatus(ctx, obj, patcher)
+		// Patch finalizers, status and conditions.
+		if err := r.finalizeStatus(ctx, obj, patcher); err != nil {
+			log.Error(err, "Status update failed")
```

Comment:
Instead of logging the error (and thereby ignoring the fact that the final persistence failed), I would append to `retErr` using [`errors.NewAggregrate`](https://pkg.go.dev/sigs.k8s.io/kind/pkg/errors#NewAggregate) so the reconciliation as a whole is retried (preventing the reflection getting lost until next interval).